intermol/gromacs/gromacs_parser.py

```diff
@@ -702,99 +670,88 @@ def create_molecule(self, top_moltype, mol_name):
+            else:
+                atom.bondingtype = atomtype_str
             if atom.mass.get(state)._value < 0:
-                if intermol_atomtype.mass._value >= 0:
-                    atom.mass = (state, intermol_atomtype.mass)
+                if atomtype.mass._value >= 0:
+                    atom.mass = (state, atomtype.mass)
                 else:
                     logger.warn("Suspicious mass parameter found for atom "
                                 "{0}. Visually inspect before using.".format(atom))
-            atom.sigma = (state, intermol_atomtype.sigma)
-            atom.epsilon = (state, intermol_atomtype.epsilon)
+            if atom.charge[state] is None:
+                if atomtype.charge is None:
+                    atom.charge = (state, 0.0 * units.elementary_charge)
+                else:
+                    atom.charge = (state, atomtype.charge)
+
+            atom.sigma = (state, atomtype.sigma)
+            atom.epsilon = (state, atomtype.epsilon)
 
         self.current_molecule.add_atom(atom)
         self.n_atoms_added += 1
 
-    def create_bond(self, bond):
+    def create_bond(self, bond_entry):
         n_atoms = 2
-        numeric_bondtype = bond[n_atoms]
-        atoms = [int(n) for n in bond[:n_atoms]]
-        btypes = tuple([self.lookup_atom_bondingtype(int(x))
-                        for x in bond[:n_atoms]])
+        atoms = [int(n) for n in bond_entry[:n_atoms]]
+        bondingtypes = tuple(self.lookup_atom_bondingtype(x) for x in atoms)
 
         # Get forcefield parameters.
-        if len(bond) == n_atoms + 1:
-            bond_type = self.find_forcetype(btypes, self.bondtypes)
-        else:
-            bond[0] = btypes[0]
-            bond[1] = btypes[1]
-            bond = " ".join(bond)
-            bond_type = self.process_forcetype(btypes, 'bond', bond, n_atoms,
-                self.gromacs_bond_types, self.canonical_bond)
-            bond = bond.split()
-
-        # Create the actual force.
-        if numeric_bondtype in self.gromacs_bonds:
-            gromacs_bond = self.gromacs_bonds[numeric_bondtype]
-            # Connection bonds don't have bondtypes.
-            if gromacs_bond == ConnectionBond:
-                kwds = dict()
-            else:
-                kwds = self.choose_parameter_kwds_from_forces(
-                    bond, n_atoms, bond_type, gromacs_bond)
-            # Give it canonical form parameters.
-            canonical_bond, kwds = self.canonical_bond(kwds, gromacs_bond,
-                                                        direction='into')
-            new_bond = canonical_bond(*atoms, **kwds)
+        if bond_entry[2] == '5':
+            bondtype = ConnectionBondType(*bondingtypes)
+        elif len(bond_entry) == n_atoms + 1:
+            bondtype = self.find_forcetype(bondingtypes, self.system.bondtypes)
         else:
-            logger.warn("Unsupported Gromacs bondtype: {0}".format(numeric_bondtype))
```

Comment:
This function clearly works a bit differently now since it now just looks for the bondtype. This specific error is caught in `process_forcetype`as far as I can tell. 

We should probably re-evaluate whether or not to through an error here. Currently, if this is determined to be an unsupported type, it will happily create a bond without a bondtype. The motivation here originally was to let a system like this to be partially converted but I'm starting to think we might want to be a bit more rigid and throw an actual exception rather than just warning the user.